src/com/boxboat/jenkins/library/config/CommonConfigBase.groovy

```diff
@@ -17,6 +17,8 @@ class CommonConfigBase<T> extends BaseConfig<T> {
 
     String vaultKey
 
+    Map<String, IUserConfig> userConfig
```

Comment:
I think `Map<String, Object>` may be more appropriate, as it would allow for arbitrary data types e.g. lists, maps, strings that do not require the interface

Reply:
Doh! Thank you, I'll update that 😄